numba/cuda/cudaimpl.py

```diff
@@ -468,6 +468,48 @@ def ptx_round(context, builder, sig, args):
     ])
 
 
+# This rounding implementation follows the algorithm used in the "fallback
+# version" of double_round in CPython 3.7.5's Objects/floatobject.c.
```

Comment:
Please add an exact reference with SHA to the implementation. Example:
https://github.com/numba/numba/blob/626b40eddc9a29a1e89bfb6b8706e4f390cfcc4a/numba/cpython/hashing.py#L30-L31